src/Microsoft.Health.Fhir.Core/Features/Operations/Export/Models/ExportJobProgress.cs

```diff
@@ -10,12 +10,15 @@ namespace Microsoft.Health.Fhir.Core.Features.Operations.Export.Models
 {
     public class ExportJobProgress
     {
-        public ExportJobProgress(string continuationToken, uint page, string resourceId = null, ExportJobProgress subSearch = null)
+        public ExportJobProgress(string continuationToken, uint page, string resourceId = null, ExportJobProgress subSearch = null, ExportJobFilter filter = null)
         {
             ContinuationToken = continuationToken;
             Page = page;
             TriggeringResourceId = resourceId;
             SubSearch = subSearch;
+            CurrentFilter = filter;
+
+            RunFilteredSearches = false;
```

Comment:
Can we set `RunFilteredSearches = true` if `filter == null`? This we can avoid the initial check in `ExportJobTask` `if (!progress.RunFilteredSearches)` where we have to do an extra `UpdateJobRecordAsync` to set this to `true`. This way we can avoid the update call for requests that do not have any filters.